newrelic/resource_newrelic_notifications_channel.go

```diff
@@ -0,0 +1,175 @@
+package newrelic
+
+import (
+	"context"
+	"fmt"
+	"github.com/newrelic/newrelic-client-go/pkg/ai"
+	"github.com/newrelic/newrelic-client-go/pkg/notifications"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+)
+
+var notificationsChannelTypes = map[notifications.AiNotificationsChannelType][]string{
+	"EMAIL":                         {},
+	"SERVICE_NOW":                   {},
+	"PAGERDUTY_ACCOUNT_INTEGRATION": {},
+	"PAGERDUTY_SERVICE_INTEGRATION": {},
+	"WEBHOOK":                       {},
+}
+
+var notificationsChannelProductTypes = map[notifications.AiNotificationsProduct][]string{
+	"ALERTS":         {},
+	"DISCUSSIONS":    {},
+	"ERROR_TRACKING": {},
+	"IINT":           {},
+	"NTFC":           {},
+	"PD":             {},
+	"SHARING":        {},
+}
+
+func resourceNewRelicNotificationChannel() *schema.Resource {
+	validNotificationChannelTypes := make([]string, 0, len(notificationsChannelTypes))
```

Comment:
Terraform offers an API for adding a `ValidateFunc` field to schema attributes.

I _think_ you might be able to utilize a pattern this example [here](https://github.com/newrelic/terraform-provider-newrelic/pull/1885/files#diff-73892bfe5b05f4ce2e8ab5063cd4cc0d1a91dc758f3feeab5e41fde9733bce7bR155-R167) and [here](https://github.com/newrelic/terraform-provider-newrelic/pull/1885/files#diff-548291a6ec014d4cb9c5a4d91e64b93a29d83f303795b54d26918ac63f24fdf1R85)

Reply:
Fixed